artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -837,16 +837,26 @@ protected synchronized void lazyScanAfterReloadPersistence() {
       messageDataScanned = MessageDataScanningStatus.SCANNED.code;
       modified = false;
 
+      makeDurable();
+   }
+
+   private void makeDurable() {
+      assert messageDataScanned == MessageDataScanningStatus.SCANNED.code;
       // Message state should reflect that is came from persistent storage which
       // can happen when moved to a durable location.  We must re-encode here to
       // avoid a subsequent redelivery from suddenly appearing with a durable header
       // tag when the initial delivery did not.
       if (!isDurable()) {
```

Comment:
Can't we just remove this?

Reply:
It would be the best possible scenario: just don't know if we have test coverage to be sure that it won't break anything...still need to investigate why we have this..idea?

Reply:
Removing would be a good option as the broker is currently altering the Delivery in a way that changes what the sender originally intended for it.  Any forwarding or resending from other intermediaries will see behavioural changes down the line because the Delivery shifted from non-durable to durable without (as far as I can tell) a valid reason for it doing so.

Reply:
@franz1981 I will do the following: mark this as DO-NOT-MERGE-YET... if you could remove these lines... run the testsuite and see if anything breaks...



I believe this was written as in.. I feel like it should have been durable.. why not set it to durable.. as opposed to it's really breaking anything.

Reply:
just completed a testsuite.. .and I see no need for this re-encoding.

I say lets get rid of this.

@tabish121 do you remember why you added this on a851a8f93f30972d252f2bff0bb3d5847cfd7b5f ?


if you're ok with removing it... it should be gone.

Reply:
Thanks guys for the help on this, highly appreciated!!! I am super happy ro follow the path that both have found and I can close this!! 
A code removed is a code we doesn't have to maintain anymore, yeeeeee

Reply:
The change I made was to stabilize what was already happening where on reload from persistence the broker tagged the durable flag in AMQPMessage as true and then depending on other factors in the message content and redelivery state it would sometimes pop out marked with a durable header and other times not.  I chose to preserve the behaviour that happened the most which was that the message reappeared with a re-encode that resulted in a Header being applied.

Reply:
@franz1981 it's safe to remove it then. amend this PR to remove it.

Reply:
The label can be removed: now the PR is removing the legacy code and improving the test coverage to spot any regression on this behaviour :+1: